contribution_checker/_commits.py

```diff
@@ -73,66 +84,100 @@ def _find_commit_matches(repo: Repo, commits: list, pattern: str) -> list:
             logging.debug(
                 "Commit %s by author '%s' matches pattern", commit["hash"], commit["email"]
             )
-            # Extract stats for this commit
-            files = []
-            added = 0
-            removed = 0
-            try:
-                diff = repo.git.diff("--numstat", f"{commit['hash']}~1", commit["hash"])
-            except exc.GitCommandError:
-                # Most likely happens if commit is the first in the repo
-                # In this case, 4b825dc642cb6eb9a060e54bf8d69288fbee4904 is empty object tree
-                diff = repo.git.diff(
-                    "--numstat", "4b825dc642cb6eb9a060e54bf8d69288fbee4904", commit["hash"]
-                )
-
-            for line in diff.split("\n"):
-                match = re.match(r"(\d+)\s+(\d+)\s+(.+)", line)
-                if match:
-                    files.append(match[3])
-                    added += int(match[1])
-                    removed += int(match[2])
-
-            commit["changes"] = {
-                "files": len(files),
-                "added": added,
-                "removed": removed,
-            }
-
-            # Append commit to list
+            commit["matched"] = True
+            # Get stats about matched commit
+            # Note: Creating these stats is resource-intense. As soon as we
+            # don't want to compare the stats of matched vs. unmatched commits,
+            # this isn't necessary
+            commit = _get_commit_stats(repo, commit)
+            # Append commit to list of matches commits
             matched_commits.append(commit)
         else:
             logging.debug(
                 "Commit %s by author '%s' does not match pattern", commit["hash"], commit["email"]
             )
+            commit["matched"] = False
+            # Append commit to list of matches commits
+            matched_commits.append(commit)
```

Comment:
Slightly confusing that a non-matched commit is added to the list of matched ones. Maybe the identifier of `matched_commits` should be updated?